Code/Source/RobotImporter/RobotImporterWidgetUtils.cpp

```diff
@@ -0,0 +1,109 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RobotImporterWidgetUtils.h"
+#include <QFileDialog>
+#include <QMessageBox>
+#include <QWidget>
+
+#include <AzCore/Utils/Utils.h>
+
+namespace ROS2::RobotImporterWidgetUtils
+{
+    namespace Internal
+    {
+        AZStd::optional<QString> QueryUserForPath(
+            const AZStd::string& extensionDescription, QFileDialog::FileMode mode, QWidget* parent = nullptr)
+        {
+            QFileDialog importFileDialog(parent);
+            importFileDialog.setDirectory(AZ::Utils::GetProjectPath().c_str());
+            importFileDialog.setFileMode(mode);
+            importFileDialog.setNameFilter(QObject::tr(extensionDescription.c_str()));
+            importFileDialog.setViewMode(QFileDialog::Detail);
+
+            int result = importFileDialog.exec();
+            if (result != QDialog::DialogCode::Accepted)
+            {
+                return AZStd::nullopt;
+            }
+
+            return importFileDialog.selectedFiles().first();
+        }
+
+        ExistingPrefabAction QueryUserForExistingPrefabAction(QWidget* parent = nullptr)
+        {
+            QMessageBox msgBox(parent);
+            msgBox.setWindowTitle(QObject::tr("Prefab file exists"));
+            msgBox.setText(QObject::tr("The prefab file already exists."));
+            msgBox.setInformativeText(QObject::tr("Do you want to overwrite it or save it with another file name?"));
+            msgBox.setStandardButtons(QMessageBox::Save | QMessageBox::Discard | QMessageBox::Cancel);
+            msgBox.setDefaultButton(QMessageBox::Discard);
+            msgBox.setButtonText(QMessageBox::Save, QObject::tr("Overwrite"));
+            msgBox.setButtonText(QMessageBox::Discard, QObject::tr("Save As..."));
+
+            switch (msgBox.exec())
+            {
+            case QMessageBox::Save:
+                return ExistingPrefabAction::Overwrite;
+            case QMessageBox::Discard:
+                return ExistingPrefabAction::CreateWithNewName;
+            default:
+                return ExistingPrefabAction::Cancel;
+            }
+        }
+    } // namespace Internal
+
+    AZStd::optional<AZStd::string> QueryUserForURDFPath(QWidget* parent)
+    {
+        std::optional<QString> path = Internal::QueryUserForPath("Unified Robot Description Format (*.urdf)", QFileDialog::ExistingFiles);
+        if (!path)
+        {
+            return AZStd::nullopt;
+        }
+
+        if (path->isEmpty())
+        {
+            QMessageBox::critical(parent, QObject::tr("Empty path provided"), QObject::tr("No path was provided. Please try again"));
+            return QueryUserForURDFPath();
+        }
+
+        if (!QFile::exists(path.value()))
+        {
+            QMessageBox::critical(parent, QObject::tr("Does not exist"), QObject::tr("Provided path does not exist. Please try again"));
+            return QueryUserForURDFPath();
+        }
+
+        return path->toStdString().c_str();
```

Comment:
`path.toUtf8().constData()` allows for use of a path with UTF-8 characters on it and avoids the need to allocate a std::string. In this scenario a QByteArray is created